VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -148,4 +122,110 @@ public void onClickAddComment(@NonNull DiscussionComment comment) {
     public void onClickViewComments(@NonNull DiscussionComment comment) {
         router.showCourseDiscussionComments(getActivity(), comment, discussionThread.isClosed());
     }
+
+    public static class DiscussionResponsesLoader implements
+            InfiniteScrollUtils.PageLoader<DiscussionComment> {
+        private InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback;
+
+        private Context context;
+        private GetCommentListTask getCommentListTask;
+        private GetCommentListTask getEndorsedListTask;
+
+        private int nextPage = 1;
+        private String threadId;
+        private boolean isQuestionTypeThread;
+
+        private ThreadComments unendorsedComments;
+        private boolean isEndorsedFetched = false;
+
+        public DiscussionResponsesLoader(@NonNull Context context,@NonNull String threadId,
+                                         boolean isQuestionTypeThread) {
+            this.context = context;
+            this.threadId = threadId;
+            this.isQuestionTypeThread = isQuestionTypeThread;
+        }
+
+        @Override
+        public void loadNextPage(@NonNull InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback) {
+            this.callback = callback;
+            if (isQuestionTypeThread && !isEndorsedFetched) {
+                getEndorsedList();
```

Comment:
Since the tasks are async, there is no guarantee that the endorsed comments will be received before the non-endorsed comments; You should not add any items to the adapter until you've received the endorsed comments.

I can see two strategies for this:
a) Run both tasks in parallel; in the onSuccess of both tasks, check to see if the other has already finished and combine the result into a single call to onPageLoaded
b) Do not run both tasks in parallel; First, run the endorsed task and wait for it to finish, calling onPageLoaded when it does (even if there are zero results). If there aren't enough endorsed comments to fill the screen, `loadNextPage` will get triggered a second time, at which point you can start loading the rest of the comments.

(b) is cleaner/simpler but (a) will usually have less latency. In either case, you don't need any kind of `onPartialPageLoaded` method